pkg/gofr/grpc.go

```diff
@@ -57,3 +58,49 @@ func (g *grpcServer) Shutdown(ctx context.Context) error {
 		return nil
 	})
 }
+
+// RegisterService adds a gRPC service to the GoFr application.
+func (a *App) RegisterService(desc *grpc.ServiceDesc, impl interface{}) {
+	a.container.Logger.Infof("registering GRPC Server: %s", desc.ServiceName)
+	a.grpcServer.server.RegisterService(desc, impl)
+
+	injectContainer(impl, a.container)
+
+	a.grpcRegistered = true
+}
+
+func injectContainer(impl any, c *container.Container) {
+	val := reflect.ValueOf(impl)
+
+	if val.Kind() != reflect.Pointer {
+		c.Logger.Debugf("cannot inject container into non-addressable implementation of `%s`, consider using pointer",
+			val.Type().Name())
+		return
+	}
+
+	val = val.Elem()
+	tVal := val.Type()
+
+	for i := 0; i < val.NumField(); i++ {
+		f := tVal.Field(i)
+		v := val.Field(i)
+
+		if f.Type == reflect.TypeOf(c) {
+			if !v.CanSet() {
+				c.Logger.Errorf("cannot inject container as it is not addressable or is fail")
+				continue
```

Comment:
You log an error, an error means you cannot continue usually.

I would have expected a break or return here then

Otherwise it means you can have more than one error when injecting the container.

Or it means you can have an error, but then a success. But there I wouldn't expect to log an error before a success, only when at the end of the loop, no solutions were found.

That's why I think I would expect break or return here instead of continue

Reply:
Yes, I think we can do a `break` / `return error` here and not start the GRPC service altogether, earlier it was supposed to be in DEBUG mode but then changed it to ERROR since the user expects the container to be populated.